custom_components/spotcast/services/play_media.py

```diff
@@ -69,13 +69,16 @@ async def async_play_media(hass: HomeAssistant, call: ServiceCall):
     if uri is None:
         pass
     elif uri.startswith("spotify:track:"):
-        uri, index = await async_track_index(account, uri)
-        LOGGER.debug(
-            "Switching context to song's album `%s`, with offset %d",
-            uri,
-            index,
-        )
-        extras["offset"] = index
+        if extras.get("track_context") == "track":
+            LOGGER.debug("Playing track `%s` only once", uri)
```

Comment:
Change the message to indicate the context used is the track only, but don't mention "only once". That would be decided by either:

- the current active mode for the account if no `repeat` was provided
- the value in `extra.repeat`

Other than that, all test passing and everything is looking good

Reply:
Perfect :-)